### PR TITLE
Run Flutter framework tests against the web engine in Cirrus

### PR DESCRIPTION
**Pull request by mdebbar**
I got bit multiple times because flutter tests currently don't run against the web engine in Cirrus (https://github.com/flutter/engine/pull/16268, https://github.com/flutter/engine/pull/14569).

These sort of mistakes could potentially be avoided too:
- https://github.com/flutter/engine/pull/16075
- https://github.com/flutter/engine/pull/15694

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -80,6 +80,10 @@ task:
       test_framework_script: |
         cd $FRAMEWORK_PATH/flutter/packages/flutter
         ../../bin/flutter test --local-engine=host_debug_unopt
+      # Run Flutter framework tests against the web engine.
+      test_framework_web_script: |
+        cd $FRAMEWORK_PATH/flutter/packages/flutter
```
Comment (nturgut) [inline .cirrus.yml]: Shall we add this to LUCI instead since Godofredo is about to finish LUCI recipes :)

I think Linux Chrome already runs!

Comment (mdebbar) [reply .cirrus.yml]: Good point. I spoke with @godofredoc and for the time-being, we are continuing to use Cirrus until the end of Q1. So I think it's still worth adding these tests to Cirrus now.

**Comment by godofredoc** [comment]
Actually LUCI solves the duplication of work, it basically builds once, archive the artifacts to isolated and then multiple tasks can be triggered to execute the tests using the same build artifacts.

**Comment by yjbanov** [comment]
> Actually LUCI solves the duplication of work, it basically builds once, archive the artifacts to isolated and then multiple tasks can be triggered to execute the tests using the same build artifacts.

Does it happen automatically? If you looks at this PR's LUCI builds, [this](https://ci.chromium.org/p/flutter/builders/try/Linux%20Web%20Engine/55) and [this](https://ci.chromium.org/p/flutter/builders/try/Linux%20Host%20Engine/3753), the "build host_debug_unopt" was fully executed on both builds, taking 4-5 minutes. So perhaps LUCI can do it, but it doesn't for us.

**Comment by godofredoc** [comment]
> > Actually LUCI solves the duplication of work, it basically builds once, archive the artifacts to isolated and then multiple tasks can be triggered to execute the tests using the same build artifacts.
> 
> Does it happen automatically? If you look at this PR's LUCI builds, [this](https://ci.chromium.org/p/flutter/builders/try/Linux%20Web%20Engine/55) and [this](https://ci.chromium.org/p/flutter/builders/try/Linux%20Host%20Engine/3753), the "build host_debug_unopt" was fully executed on both builds, taking 4-5 minutes. So perhaps LUCI can do it, but it doesn't for us.

That is correct luci can optimize to that level but we need to tell it to do for us. For example for linux web engine we are planning to build host_debug_unopt once, archive to isolated and then trigger two tasks one for firefox and one for chrome using the same artifacts.

For post submit we can optimize even more building once, archiving to gcs and then sharding all the tests using the gcs artifacts.

**Comment by mdebbar** [comment]
@godofredoc this PR adds 8 new tasks to Cirrus. Are there any concerns regarding our Cirrus capacity?

**Comment by yjbanov** [comment]
> @godofredoc this PR adds 8 new tasks to Cirrus. Are there any concerns regarding our Cirrus capacity?

Since these are the exact same tests we already run on every PR in flutter/flutter, this should only add engine commit rate to the existing commit rate, so I expect the incremental cost is 10%-20%.

**Pull request merged**